fix: autoclose resolved deprecated issues

hassan23 opened the pull request:
This PR resolves the following:
If there is `Dependency deprecation warning` `issue` in your repo and afterward you remove  
 that `dependency` from `package.json`, then the issue will get automatically closed.

Closes #3598  <!-- Ideally each PR should be closing an open issue -->

rarkins commented on lib/workers/repository/process/deprecated.js:
```diff
@@ -53,5 +63,23 @@ async function raiseDeprecationWarnings(config, packageFiles) {
+    const titleMatch = 'Dependency deprecation warning';
+    if (issueList && issueList.length) {
+      const depricatedIssues = issueList.filter(i =>
+        i.title.includes(titleMatch)
```
startsWith

rarkins commented on lib/workers/repository/process/deprecated.js:
```diff
@@ -53,5 +63,23 @@ async function raiseDeprecationWarnings(config, packageFiles) {
+      );
+      for (const i of depricatedIssues) {
+        if (
+          i.state === 'open' &&
```
add this to the filter

rarkins commented:
@hassan23 what if you just build an array of the issue titles during the first (existing) loop and then use that to filter out the open issues list?

hassan23 commented:
@rarkins, please review

rarkins commented on lib/workers/repository/process/deprecated.js:
```diff
@@ -53,5 +56,20 @@ async function raiseDeprecationWarnings(config, packageFiles) {
       }
     }
+    logger.debug(
+      'Closing existing depricated package issues missing in current deprecatedPackages'
```
This should say "Checking for" not "closing". Otherwise it implies that you are closing when most times you probably are not. Also depricated => deprecated

rarkins commented on lib/workers/repository/process/deprecated.js:
```diff
@@ -53,5 +56,20 @@ async function raiseDeprecationWarnings(config, packageFiles) {
+    const issueList = await platform.getIssueList();
+    const titleMatch = 'Dependency deprecation warning';
+    if (issueList && issueList.length) {
+      const depricatedIssues = issueList.filter(
```
spelling => deprecated

rarkins commented on lib/workers/repository/process/deprecated.js:
```diff
@@ -53,5 +56,20 @@ async function raiseDeprecationWarnings(config, packageFiles) {
+      'Closing existing depricated package issues missing in current deprecatedPackages'
+    );
+    const issueList = await platform.getIssueList();
+    const titleMatch = 'Dependency deprecation warning';
```
Find a way to share this string with the earlier code for creating issues that also uses the same (line 44)

hassan23 commented:
@rarkins please review

rarkins commented:
Code looks good. @hassan23 have you tested this last commit against a "real" repository? I would like to test it too before merging. Do you have an example deprecated package you suggest I test with?

hassan23 commented:
> Code looks good. @hassan23 have you tested this last commit against a "real" repository? I would like to test it too before merging. Do you have an example deprecated package you suggest I test with?

I have tested this on a test repo, I used `nomnom` npm package which is deprecated now,
Let me try this one more time on a real repo, and get back to you in 15 minutes

hassan23 commented:
> Code looks good. @hassan23 have you tested this last commit against a "real" repository? I would like to test it too before merging. Do you have an example deprecated package you suggest I test with?

Steps I followed to check the functionality on real repo:
1. Added `nomnom` npm package in `package.json`
2. ran `yarn start username/reponame`
3. An issue gets opened by the bot with title `Dependency deprecation warning: nomnom (npm)`
3. Removed `nomnom` npm package from `package.json`
4. Issue opened in step 2 gets automatically closed
